manifests/init.pp

```diff
@@ -1,10 +1,21 @@
-# == Class: lvm
-#
 class lvm (
-  $volume_groups = {},
-) {
+  $manage_package = $::lvm::params::manage_package,
+  $package_name   = $::lvm::params::package_name,
+  $package_ensure = $::lvm::params::package_ensure,
+  $mountpoints    = $::lvm::params::mountpoints,
+  $volume_groups  = $::lvm::params::volume_groups,
+) inherits ::lvm::params {
 
   validate_hash($volume_groups)
+  validate_hash($mountpoints)
+  validate_bool($manage_package)
+  validate_string($package_name)
 
-  create_resources('lvm::volume_group', $volume_groups)
+  contain '::lvm::install'
+  contain '::lvm::config_mountpoints'
+  contain '::lvm::config_volumes'
```

Comment:
Because this module must still support < 3.5, we can't use contain. Could you change this to the anchor pattern?